### PR TITLE
15050 consent continuation out

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#15050

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-schemas license (Apache 2.0).

### CODE REVIEW

File: src/registry_schemas/schemas/consent_continuation_out.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://bcrs.gov.bc.ca/.well_known/schemas/consent_continuation_out",
+    "title": "Registrars Notation Information Schema",
```
Comment (argush3) [inline src/registry_schemas/schemas/consent_continuation_out.json:4]: title needs to be updated

**Comment by argush3** [comment]
In the designs, under the documents delivery section there is an optional client email address beside the completing party label.  I can't remember if we had said anything specific about this during backlog refinement.  do we need to account for this in the schema?

image

**Comment by vysakh-menon-aot** [comment]
> In the designs, under the documents delivery section there is an optional client email address beside the completing party label. I can't remember if we had said anything specific about this during backlog refinement. do we need to account for this in the schema?
> 
> image

This is part of header (called `documentOptionalEmail`)

**Review by argush3 — APPROVED** [review]

File: src/registry_schemas/example_data/schema_data.py [hunk 2]
```diff
@@ -1210,6 +1210,11 @@
     'fileKey': '011e332d-1b8e-4218-8710-ad8ac1fbc592.pdf'
 }
 
+CONSENT_CONTINUATION_OUT = {
+    'orderDetails': 'A note to explain the consent to continue out',
+    'courtOrder': COURT_ORDER
```
Comment (severinbeauvais) [inline src/registry_schemas/example_data/schema_data.py:1215]: Where are we storing the Consent's expiry date?

File: tests/unit/test_consent_continuation_out.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    if errors:
+        for err in errors:
+            print(err.message)
+    print(errors)
+
+    assert is_valid
+
+
+def test_consent_continuation_out_invalid_schema():
```
Comment (severinbeauvais) [inline tests/unit/test_consent_continuation_out.py:36]: What about validation for whether the business is Active and in Good Standing?

What about validation for "only staff can file this"?

Comment (severinbeauvais) [reply tests/unit/test_consent_continuation_out.py:36]: Sorry, I just realized THIS is the schema change, and the Legal API will handle the business validation. Please resolve this conversation if you want.

File: src/registry_schemas/example_data/schema_data.py (hunk 2 above)
Comment (severinbeauvais) [reply src/registry_schemas/example_data/schema_data.py:1215]: ^^ The UI will not provide the expiry date -- the BE will compute it upon filing.

Comment (vysakh-menon-aot) [reply src/registry_schemas/example_data/schema_data.py:1215]: Yeah! As we discussed I will create a new filed in business table. Thats not part of schema

**Review by severinbeauvais — APPROVED** [review]

File: src/registry_schemas/schemas/consent_continuation_out.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    ],
+    "type": "object",
+    "properties": {
+        "consentContinuationOut": {
+            "type": "object",
+            "required": [
+                "orderDetails"
+            ],
+            "properties": {
+                "orderDetails": {
```
Comment (thorwolpert) [inline src/registry_schemas/schemas/consent_continuation_out.json:17]: Is this an _order_, or a note from the Registry?
It doesn't seem similar to a Registrar's Order, and it's not a _courtOrder_

**Review by thorwolpert — COMMENTED** [review]
I need a bit more information on the embedded order element.

File: src/registry_schemas/schemas/consent_continuation_out.json (hunk 4 above)
Comment (vysakh-menon-aot) [reply src/registry_schemas/schemas/consent_continuation_out.json:17]: https://preview.uxpin.com/f3e9c0da57601f193d2f0c6b1946134b2ddeb2fe#/pages/160490823/comment/sitemap

Comment (vysakh-menon-aot) [reply src/registry_schemas/schemas/consent_continuation_out.json:17]: It's similar to Registrar's Order

Comment (vysakh-menon-aot) [reply src/registry_schemas/schemas/consent_continuation_out.json]: Created a generic schema called `order` and referred in registrars order, registrars notation and consent continuation out

File: src/registry_schemas/schemas/consent_continuation_out.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    "required": [
+        "consentContinuationOut"
+    ],
+    "type": "object",
+    "properties": {
+        "consentContinuationOut": {
+            "$ref": "https://bcrs.gov.bc.ca/.well_known/schemas/order"
+        }
+    }
+}
```
Comment (severinbeauvais) [inline src/registry_schemas/schemas/consent_continuation_out.json:15]: Is this schema referencing itself?

File: src/registry_schemas/example_data/schema_data.py [hunk 6]
```diff
@@ -1210,6 +1210,12 @@
     'fileKey': '011e332d-1b8e-4218-8710-ad8ac1fbc592.pdf'
 }
 
+CONSENT_CONTINUATION_OUT = {
+    'orderDetails': 'A note to explain the consent to continue out',
+    'fileNumber': '#1234-5678/90',
+    'effectOfOrder': 'planOfArrangement'
+}
```
Comment (severinbeauvais) [inline src/registry_schemas/example_data/schema_data.py:1217]: What about `courtOrder`?

File: src/registry_schemas/schemas/order.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            "maxLength": 500,
+            "description": "planOfArrangement"
+        },
+        "orderDetails": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 2000,
+            "description": "A brief note to explain the purpose of the Order."
+        }
+    }
```
Comment (severinbeauvais) [inline src/registry_schemas/schemas/order.json:35]: Should `courtOrder` be in here? If not, where is it?

File: src/registry_schemas/example_data/schema_data.py (hunk 6 above)
Comment (vysakh-menon-aot) [reply src/registry_schemas/example_data/schema_data.py:1217]: `courtOrder` filing has a different required prop (`fileNumber`) which I cannot specify outside schema. Thats why we created separate schema

File: src/registry_schemas/schemas/order.json (hunk 7 above)
Comment (severinbeauvais) [reply src/registry_schemas/schemas/order.json:35]: Answered by comment above.

Comment (vysakh-menon-aot) [reply src/registry_schemas/schemas/order.json:35]: No. This include those prop's as same as registrars order, registrars notation

File: src/registry_schemas/example_data/schema_data.py (hunk 6 above)
Comment (vysakh-menon-aot) [reply src/registry_schemas/example_data/schema_data.py:1217]: Yeah! This is what I propose. Will wait for @thorwolpert feedback.

File: src/registry_schemas/schemas/order.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            "format": "date-time",
+            "examples": ["1970-01-01T00:00:00+00:00"]
+        },
+        "effectOfOrder": {
+            "type": "string",
+            "minLength": 0,
+            "maxLength": 500,
+            "description": "planOfArrangement"
+        },
+        "orderDetails": {
```
Comment (severinbeauvais) [inline src/registry_schemas/schemas/order.json:29]: I believe that, for some filings, the UI calls this property, `details`.

1. https://github.com/bcgov/business-filings-ui/blob/c9afa338ea6b4c47d691a22b4b752cc59733710a/src/components/dialogs/AddStaffNotationDialog.vue#L337
2. https://github.com/bcgov/business-filings-ui/blob/c9afa338ea6b4c47d691a22b4b752cc59733710a/src/components/dialogs/AddStaffNotationDialog.vue#L350
3. https://github.com/bcgov/business-filings-ui/blob/c9afa338ea6b4c47d691a22b4b752cc59733710a/src/components/dialogs/AddStaffNotationDialog.vue#L380

Comment (vysakh-menon-aot) [reply src/registry_schemas/schemas/order.json]: I didn't realize we have more places with these. Reverting this for now

**Review by thorwolpert — APPROVED** [review]
Thanks for the explanation and the field name change.

**Pull request merged**